### PR TITLE
Just biodiesel recipe

### PR DESCRIPTION
**Pull request by kevingranade**
DLaboratory closed their PR, possibly because they didn't like the feedback?  Not sure.
Regardless, I'd like the recipe at least to make it in.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
The balancing could use some tuning and not just fine-tuning.

Current techno-magic recipe turns 6 fluid sacs into gasoline, diesel and lamp oil (one stack each).
The recipe here turns no less than 40 volume of base materials into a single unit of just diesel. That is, a single cup.

Even when producing diesel from vodka, it would be: 400 units of vodka (volume 57)->200 ethanol (volume 1 or 200, not sure if I'm reading it right)->800 diesel (volume 4).
400 units of strong alcohol is more than a typical liquor store has in stock. 400 units of a single type of strong alcohol is more than 99% of players will ever see.

**Comment by kevingranade** [comment]
Hrm yes, the original was probably based on that dumb reality thing we keep hearing about.
I'm strongly tempted to pull both toward some middle ground rather than just making the biodiesel recipe worse.

**Comment by kevingranade** [comment]
Some quick research indicates that the yield for pure vegetable oil is quite high by volume, something like 75% if I'm reading correctly (source: http://makezine.com/projects/make-03/backyard-biodiesel/), and the ethanol will be at something like a 1/4 ratio with the oil (unless it's supposed to act as both methanol/methoxy AND ethanol, at which point I'm not sure what the ratio should be) So roughly speaking for 20 units of lard there should be 5 units of ethanol, and it would yield something like 15 units of biodiesel.
If the ethanol is standing in for both ethanol and methanol, then any portion of it acting as ethanol will be included in the final volume, unlike the methanol which must be removed.
Fluid sacs are supposed to be very nearly fuel oil to start with (yet somehow still edible, wot?), so now that I've looked at the biodiesel process again, I think leaving it at its current yield is fine.

**Comment by kevingranade** [comment]
Totally rebalanced, it is now:
2 volume of hard liquor = 1 volume water, 1 volume ethanol.
20 volume dense fat, or much more adulterated fat + 5 vol ethanol + 5 vol methanol (with ethanol standing in for it) + a pinch of lye + 1 vol water (washing) = 20 vol diesel (25% of the oil volume lost, but replaced with ethanol).
It's possibly too far in the other direction now, it's quite a lot of it.

**Comment by Coolthulhu** [comment]
The batch size could easily be cut by a factor of 10. That way it would be craftable even in summer without having to rely on tricks like rotten fat not disappearing except on map load or having to preserve everything. Plus, it would allow using more of the feedstock, not just the ingredients that can be amassed above the required threshold.
1 lye is literally a pinch - it is produced in batches of 200. Multiplying the usage by 10 (required for the above batch size change) would not affect balance. It is used more like tool than a component here.

Is there any reason why canola seeds are used directly rather than using cooking oil or is it just because that how it was in the original recipe?
Could lamp oil be used in biodiesel production?

Can biodiesel actually be produced from flesh and organs?
If flesh can be used, then fish flesh could probably be used too.
And probably also goo blobs - they have same nutrition as raw meat.

I recall there being a convention that unprocessed tainted fat is worth 1/2 of regular fat in recipes that allow it, such as candles. For example, tainted tallow requires 4 tainted fat. Here they're equal.

**Comment by kevingranade** [comment]
Agree with the batch size adjustment, replaced canola seeds with oil, just didn't notice it there.
No idea if lamp oil would be usable, likewise I'm pulling the meats, if they can have fat rendered out of them, it's probably best as a separate step.
Also cut the efficacy of the tainted materials.

**Comment by Cat09** [comment]
side note: a type of N-gas + fertilizer can be produced from any organic material in an anaerobic digester tank, could be useful for fuel from byproducts / rotting junk.

File: data/json/recipes/recipe_medsandchemicals.json [hunk 1]
```diff
@@ -2609,80 +2609,108 @@
...
+    "autolearn": false,
+    "book_learn": [["textbook_biodiesel", 5]],
+    "tools": [[[ "chemistry_set", 10 ]]],
+    "components": [
+        [[ "water_clean", 1 ], [ "water", 1 ]],
+        [[ "chem_ethanol", 1 ]],
+        [[ "cooking_oil", 2 ],
+         [ "lard", 2 ],
+         [ "tallow", 2 ],
+         [ "tallow_tainted", 4 ],
```
Comment (Coolthulhu) [inline data/json/recipes/recipe_medsandchemicals.json]: Minor: tallow_tainted is made from 4 fat_tainted.

**Comment by Coolthulhu** [comment]
Minor:
* Tainted tallow takes 2x as much tainted fat as tallow does take regular fat, so the doubling on the recipe makes using tainted tallow 2x as costly as using tainted fat
* Diesel production only takes the chem set, no qualities such as boiling or containing
* No batch modifier means that producing significant quantities of ethanol takes a lot of time (20 minutes per unit)
* Tiny base time (rather than batch modifier) means 2 units of diesel can be produced in no time (3 minutes, compare with clean water's 5 minutes)

For the future:
* Recipes that use alcohols at the moment can't use ethanol yet. Disinfectant, bandages, molotovs etc.

**Comment by DLaboratory** [comment]
Sorry. I closed that old PR because I think the idea of adding biodiesel, its textbook and engines to the game might be really not thoughtful. Then I closed the old PR, then planned to create a new branch and write biodiesel recipe only, but I was traveling in past 2 weeks. Sorry for my delay on working this and thank you for adding it back to game!

**Pull request merged**